src/core/function_admin.tf

```
@@ -0,0 +1,398 @@
+
+      SUBSCRIPTIONS_FEED_TABLE          = "SubscriptionsFeedByDay"
+      SubscriptionFeedStorageConnection = data.azurerm_storage_account.api.primary_connection_string
+
+      // table for saving failed user data processing requests
+      FAILED_USER_DATA_PROCESSING_TABLE         = "FailedUserDataProcessing"
+      FailedUserDataProcessingStorageConnection = data.azurerm_storage_account.api.primary_connection_string
+
+      # SECRETS
+      LOGOS_URL = data.azurerm_key_vault_secret.fn_admin_ASSETS_URL.value
+
+      AZURE_SUBSCRIPTION_ID = data.azurerm_key_vault_secret.fn_admin_AZURE_SUBSCRIPTION_ID.value
+
+      ADB2C_TENANT_ID            = data.azurerm_key_vault_secret.adb2c_TENANT_NAME.value
+      ADB2C_CLIENT_ID            = data.azurerm_key_vault_secret.devportal_CLIENT_ID.value
+      ADB2C_CLIENT_KEY           = data.azurerm_key_vault_secret.devportal_CLIENT_SECRET.value
+      ADB2C_TOKEN_ATTRIBUTE_NAME = data.azurerm_key_vault_secret.adb2c_TOKEN_ATTRIBUTE_NAME.value
+
+      SERVICE_PRINCIPAL_CLIENT_ID = data.azurerm_key_vault_secret.ad_APPCLIENT_APIM_ID.value
+      SERVICE_PRINCIPAL_SECRET    = data.azurerm_key_vault_secret.ad_APPCLIENT_APIM_SECRET.value
+      SERVICE_PRINCIPAL_TENANT_ID = data.azurerm_key_vault_secret.common_AZURE_TENANT_ID.value
+
+      PUBLIC_API_KEY = data.azurerm_key_vault_secret.apim_IO_GDPR_SERVICE_KEY.value
+
+      SESSION_API_KEY = data.azurerm_key_vault_secret.app_backend_PRE_SHARED_KEY.value
+
+      __DISABLED__SENDGRID_API_KEY = data.azurerm_key_vault_secret.common_SENDGRID_APIKEY.value
+      MAILUP_USERNAME              = data.azurerm_key_vault_secret.common_MAILUP_USERNAME.value
+      MAILUP_SECRET                = data.azurerm_key_vault_secret.common_MAILUP_SECRET.value
+    }
+  }
+}
+
+# Admin resouce group
+resource "azurerm_resource_group" "admin_rg" {
+  name     = format("%s-admin-rg", local.project)
+  location = var.location
+
+  tags = var.tags
+}
+
+# Subnet to host admin function
+module "admin_snet" {
+  source                                         = "git::https://github.com/pagopa/azurerm.git//subnet?ref=v1.0.51"
+  name                                           = format("%s-admin-snet", local.project)
+  address_prefixes                               = var.cidr_subnet_fnadmin
+  resource_group_name                            = data.azurerm_resource_group.vnet_common_rg.name
+  virtual_network_name                           = data.azurerm_virtual_network.vnet_common.name
+  enforce_private_link_endpoint_network_policies = true
+
+  service_endpoints = [
+    "Microsoft.Web",
+    "Microsoft.AzureCosmosDB",
+    "Microsoft.Storage",
+  ]
+
+  delegation = {
+    name = "default"
+    service_delegation = {
+      name    = "Microsoft.Web/serverFarms"
+      actions = ["Microsoft.Network/virtualNetworks/subnets/action"]
+    }
+  }
+}
+
+module "function_admin" {
+  source = "git::https://github.com/pagopa/azurerm.git//function_app?ref=v3.4.0"
```

```suggestion
  source = "git::https://github.com/pagopa/azurerm.git//function_app?ref=v3.9.2"
```
use latest version and add:
- domain
- enable alert
- action group